### ISSUE TITLE
loss NAN problem

### ISSUE THREAD

**Issue by ltcs11** [deepinsight/insightface#387]
i am using the --ce_loss option to get the loss output while doing mobilefacenet trainging
but the output loss results  are always nan

did anyone meet with the same problem?
how should i fix this?

(i'm using batchsize 128 and training from scratch)

**Comment by nttstar** [comment]
mobilefacenet requires training by softmax loss firstly.

**Comment by elogi** [comment]
> mobilefacenet requires training by softmax loss firstly.

Situation：using --ce_loss option to train mobilefacenet, **already pretrained** by softmax；
Problem：“lossvalue“ was just fine at beginning，but turn to “nan” after few batchs；
Have you ever met this before?  
Any advice and suggestions will be greatly appreciated！

INFO:root:Epoch[0] Batch [7320]	Speed: 932.96 samples/sec	acc=0.000293	lossvalue=48.592317
INFO:root:Epoch[0] Batch [7340]	Speed: 866.37 samples/sec	acc=0.000879	lossvalue=48.279865
INFO:root:Epoch[0] Batch [7360]	Speed: 883.81 samples/sec	acc=0.000684	lossvalue=48.342774
INFO:root:Epoch[0] Batch [7380]	Speed: 919.57 samples/sec	acc=0.000879	lossvalue=48.276679
INFO:root:Epoch[0] Batch [7400]	Speed: 869.82 samples/sec	acc=0.000781	lossvalue=48.066068
INFO:root:Epoch[0] Batch [7420]	Speed: 866.56 samples/sec	acc=0.000879	lossvalue=47.864434
INFO:root:Epoch[0] Batch [7440]	Speed: 899.07 samples/sec	acc=0.000488	lossvalue=47.974631
INFO:root:Epoch[0] Batch [7460]	Speed: 893.42 samples/sec	acc=0.001172	lossvalue=48.014879
INFO:root:Epoch[0] Batch [7480]	Speed: 896.43 samples/sec	acc=0.000488	lossvalue=nan
INFO:root:Epoch[0] Batch [7500]	Speed: 900.82 samples/sec	acc=0.000781	lossvalue=nan
INFO:root:Epoch[0] Batch [7520]	Speed: 879.84 samples/sec	acc=0.001270	lossvalue=47.803672
INFO:root:Epoch[0] Batch [7540]	Speed: 921.10 samples/sec	acc=0.000879	lossvalue=47.625753
INFO:root:Epoch[0] Batch [7560]	Speed: 886.88 samples/sec	acc=0.001563	lossvalue=nan
INFO:root:Epoch[0] Batch [7580]	Speed: 887.25 samples/sec	acc=0.000586	lossvalue=47.580973
INFO:root:Epoch[0] Batch [7600]	Speed: 920.13 samples/sec	acc=0.001074	lossvalue=nan
INFO:root:Epoch[0] Batch [7620]	Speed: 909.07 samples/sec	acc=0.002246	lossvalue=nan
INFO:root:Epoch[0] Batch [7640]	Speed: 891.21 samples/sec	acc=0.001465	lossvalue=47.469683
INFO:root:Epoch[0] Batch [7660]	Speed: 918.16 samples/sec	acc=0.001563	lossvalue=nan
INFO:root:Epoch[0] Batch [7680]	Speed: 879.63 samples/sec	acc=0.001172	lossvalue=nan
INFO:root:Epoch[0] Batch [7700]	Speed: 918.21 samples/sec	acc=0.001563	lossvalue=nan
INFO:root:Epoch[0] Batch [7720]	Speed: 915.12 samples/sec	acc=0.001660	lossvalue=nan
INFO:root:Epoch[0] Batch [7740]	Speed: 865.96 samples/sec	acc=0.001465	lossvalue=47.008009
INFO:root:Epoch[0] Batch [7760]	Speed: 929.95 samples/sec	acc=0.001855	lossvalue=47.041194
INFO:root:Epoch[0] Batch [7780]	Speed: 895.16 samples/sec	acc=0.001953	lossvalue=46.928610
INFO:root:Epoch[0] Batch [7800]	Speed: 892.56 samples/sec	acc=0.002539	lossvalue=46.676923
INFO:root:Epoch[0] Batch [7820]	Speed: 898.11 samples/sec	acc=0.002734	lossvalue=nan
INFO:root:Epoch[0] Batch [7840]	Speed: 893.91 samples/sec	acc=0.001758	lossvalue=47.075309
INFO:root:Epoch[0] Batch [7860]	Speed: 865.69 samples/sec	acc=0.002051	lossvalue=46.828490
INFO:root:Epoch[0] Batch [7880]	Speed: 886.07 samples/sec	acc=0.002246	lossvalue=46.398229
INFO:root:Epoch[0] Batch [7900]	Speed: 870.09 samples/sec	acc=0.003027	lossvalue=46.901641
INFO:root:Epoch[0] Batch [7920]	Speed: 904.38 samples/sec	acc=0.002539	lossvalue=46.403756
INFO:root:Epoch[0] Batch [7940]	Speed: 927.21 samples/sec	acc=0.003320	lossvalue=nan
INFO:root:Epoch[0] Batch [7960]	Speed: 898.31 samples/sec	acc=0.003516	lossvalue=nan
INFO:root:Epoch[0] Batch [7980]	Speed: 887.01 samples/sec	acc=0.003711	lossvalue=nan
lr-batch-epoch: 0.01 7999 0
INFO:root:Epoch[0] Batch [8000]	Speed: 886.48 samples/sec	acc=0.003906	lossvalue=nan
INFO:root:Epoch[0] Batch [8020]	Speed: 884.92 samples/sec	acc=0.003027	lossvalue=45.867597
INFO:root:Epoch[0] Batch [8040]	Speed: 899.66 samples/sec	acc=0.003516	lossvalue=46.145202
INFO:root:Epoch[0] Batch [8060]	Speed: 898.91 samples/sec	acc=0.003613	lossvalue=nan
INFO:root:Epoch[0] Batch [8080]	Speed: 890.17 samples/sec	acc=0.004004	lossvalue=nan
INFO:root:Epoch[0] Batch [8100]	Speed: 890.00 samples/sec	acc=0.003711	lossvalue=nan
INFO:root:Epoch[0] Batch [8120]	Speed: 905.05 samples/sec	acc=0.003906	lossvalue=nan
INFO:root:Epoch[0] Batch [8140]	Speed: 867.92 samples/sec	acc=0.003027	lossvalue=nan

**Comment by nttstar** [comment]
try increasing the batch-size to 128*4

**Comment by ltcs11** [comment]
the same problem as @elogi  
after using **the pretrained model with softmax loss** （get training acc@0.17, lfw acc@96%, loss@8.17）
i changed the loss into arcface and begin to fine tune， then i meet with the same problem
that the training acc@0.001(become super low compare with softmax training) and loss@nan soon
and the lfw acc converge at 97%

**Comment by seanM29** [comment]
have you solved this problem?
@ltcs11

**Comment by rocinant** [comment]
It seems that I got similar issue:
after using the pretrained model with softmax loss（get training acc@0.94, lfw acc@94%）
i changed the loss into arcface and begin to fine tune, then i got training acc@0.07(super low compared with softmax training), it is still going but I don't see improvement.

i used the batch-size of 256 (max already)

**Comment by deepwilson** [comment]
Any update guys ?

**Comment by feilong** [comment]
In my case it was because the `logit` for the true class was too small and the probability after softmax became 0 due to precisions. Changing `s` to a smaller value (e.g., from 64 to 32) solved the issue for me.
